examples/chef/common/chef-operational-state-delegate-impl.cpp

```diff
@@ -201,6 +205,89 @@ void OperationalState::Shutdown()
     }
 }
 
+chip::Protocols::InteractionModel::Status chefOperationalStateWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                            const EmberAfAttributeMetadata * attributeMetadata,
+                                                                            uint8_t * buffer)
+{
+    chip::Protocols::InteractionModel::Status ret = chip::Protocols::InteractionModel::Status::Success;
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(gOperationalStateInstance != nullptr);
+    chip::AttributeId attributeId = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::OperationalState::Attributes::CurrentPhase::Id: {
+        uint8_t m = static_cast<uint8_t>(buffer[0]);
+        DataModel::Nullable<uint8_t> aPhase(m);
+        CHIP_ERROR err = gOperationalStateInstance->SetCurrentPhase(aPhase);
+        if (CHIP_NO_ERROR == err)
+        {
+            break;
+        }
+        ret = chip::Protocols::InteractionModel::Status::ConstraintError;
+        ChipLogError(DeviceLayer, "Invalid Attribute Update status: %" CHIP_ERROR_FORMAT, err.Format());
+    }
+    break;
+    case chip::app::Clusters::OperationalState::Attributes::OperationalState::Id: {
+        uint8_t currentState = gOperationalStateInstance->GetCurrentOperationalState();
+        uint8_t m            = static_cast<uint8_t>(buffer[0]);
+        CHIP_ERROR err       = gOperationalStateInstance->SetOperationalState(m);
+
+        if (currentState == to_underlying(OperationalState::OperationalStateEnum::kStopped) &&
+            m == to_underlying(OperationalState::OperationalStateEnum::kRunning))
+        {
+            gOperationalStateDelegate->mCountDownTime.SetNonNull(
+                static_cast<uint32_t>(gOperationalStateDelegate->kExampleCountDown));
```

Comment:
What does setting this timer get you? If you are expecting that after 30 seconds if goes back to stop, I don't think that works since you have not called `DeviceLayer::SystemLayer().StartTimer(System::Clock::Seconds16(1), onOperationalStateTimerTick, this)`

Reply:
Thanks for pointing this out. Added.